v1/ast/oracle/oracle.go

```diff
@@ -84,39 +86,54 @@ func (o *Oracle) FindDefinition(q DefinitionQuery) (*DefinitionQueryResult, erro
 	// references to imports or other rules. This handles intra-module, intra-package,
 	// and inter-package references.
 	for i := len(stack) - 1; i >= 0; i-- {
-		if term, ok := stack[i].(*ast.Term); ok {
-			if ref, ok := term.Value.(ast.Ref); ok {
-				prefix := ref.ConstantPrefix()
-				if rules := compiler.GetRulesExact(prefix); len(rules) > 0 {
-					return &DefinitionQueryResult{rules[0].Location}, nil
-				}
-				for _, imp := range parsed.Imports {
-					if path, ok := imp.Path.Value.(ast.Ref); ok {
-						if prefix.HasPrefix(path) {
-							return &DefinitionQueryResult{imp.Path.Location}, nil
-						}
-					}
-				}
+		term, ok := stack[i].(*ast.Term)
+		if !ok {
+			continue
+		}
+
+		ref, ok := term.Value.(ast.Ref)
+		if !ok {
+			continue
+		}
+
+		prefix := ref.ConstantPrefix()
+
+		if rules := compiler.GetRules(ref); len(rules) > 0 {
```

Comment:
GetRulesExact -> GetRules is the functional change here. The rest is just my tidying up to understand what was going on.